addons/binding/org.openhab.binding.orvibo/src/main/java/org/openhab/binding/orvibo/internal/OrviboHandlerFactory.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.orvibo.internal;
+
+import static org.openhab.binding.orvibo.OrviboBindingConstants.THING_TYPE_S20;
+
+import java.util.Collections;
+import java.util.Set;
+
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandlerFactory;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.openhab.binding.orvibo.handler.S20Handler;
+import org.osgi.service.component.ComponentContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.tavalin.s20.S20Client;
+
+/**
+ * The {@link OrviboHandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author Daniel Walters - Initial contribution
+ */
+public class OrviboHandlerFactory extends BaseThingHandlerFactory {
+
+    private final Logger logger = LoggerFactory.getLogger(OrviboHandlerFactory.class);
+    private final static Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = Collections.singleton(THING_TYPE_S20);
+    private S20Client s20Client;
```

Comment:
Hm, it looks as if your discovery service also expects the connection to be established already - so putting this code in one of two services, where you do not know when and in which order they are activated, does not look like a good idea.

The better place would be a bundle activator, so that it is always ensured that this code has been executed, before any service comes up. 
The best way would be to only connect-on-demand, i.e. if the library would support checking if it is already initialized. Then you could do it only where you really need it, i.e. in the discovery and within the ThingHandler (not the ThingHandlerFactory).